vm/devices/user_driver/src/lockmem.rs

```diff
@@ -164,12 +130,10 @@ impl crate::vfio::VfioDmaBuffer for LockedMemorySpawner {
     /// Restore mapped DMA memory at the same physical location after servicing.
     fn restore_dma_buffer(
         &self,
-        addr: u64,
         len: usize,
         _pfns: &[u64],
     ) -> anyhow::Result<crate::memory::MemoryBlock> {
-        Ok(crate::memory::MemoryBlock::new(LockedMemory::restore(
-            addr, len,
-        )?))
+        // TODO: With the recent change it may not work. Review.
+        Ok(crate::memory::MemoryBlock::new(LockedMemory::new(len)?))
```

Comment:
We need to just fail here, right? Because we can't restore the requested GPAs.

Reply:
Yes. Is it ok to do it in two waves?
1 - Fail here
2 - Return an error and then reinitialize everything as on 1st boot?

E.g. will we prioritize VM health over error discovery? And I mean to defer #2 as future enhancement.

Reply:
(ignore the link, that is just bullet point)

Reply:
I am still waiting for this path to become a failure.

Reply:
Done